.github/workflows/codeql-analysis.yml

```
@@ -0,0 +1,35 @@
+
+    permissions:
+      contents: read           # Required by the CodeQL Action
+      security-events: write   # Required to post findings
+
+    strategy:
+      matrix:
+        language: [ "javascript", "typescript" ] # Adjust to your repo's languages
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
```

actionlint (1.7.10)

[error] the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)